### DISCUSSION TITLE
Livewire does not work in non root domain (in subdirectory)

### DISCUSSION THREAD

**Discussion by lichtner** [livewire/livewire#8508] [7. Bugs (archived)]
### Livewire version

v3.5.0

### Laravel version

v11.8.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

Livewire does not work in subdirectory like https://example.com/shop 

It is possible to hack it, but this should work by default.

### Steps:

1. create new laravel app "project" + install livewire.
2. in project dir create sym link: `ln -s public shop`
3. add Counter component from livewire doc https://livewire.laravel.com/docs/quickstart
4. in `routes/web.php` add route `Route::get('/counter', Counter::class);`
5. in `bootstrap/app.php` add

```
    ->withMiddleware(function (Middleware $middleware) {
        $middleware->validateCsrfTokens(except: [
            'livewire/*',
        ]);
    })
```

6. set apache virtual host (or any other web server) to run laravel project in subdirectory, for Apache e.g:

```
<VirtualHost *:80>
    DocumentRoot    /home/www/project
    ServerName      project
    <Directory "/home/www/project/shop">
     ..... default laravel conf
   </DIrectory>
</VirtualHost>
```

7. in browser open http://project/shop (this open default laravel page)
8. but buttons in http://project/public/counter does not work because livewire ignore subdirectory "shop".

Yes, it is possible to "hack" it and add in `routes/web.php` this:

```
Livewire::setUpdateRoute(static function ($handle) {
    return Route::post('/shop/livewire/update', $handle)
        ->middleware('api');
});

Livewire::setScriptRoute(static function ($handle) {
    return Route::get('/shop/livewire/livewire.js', $handle)->name('get-route');
});

```

this works by accident but

e.g. if I name get route and display it in `counter.blade.php` like this: `{{ route('get-route') }}` 
I see `http://project/shop/shop/livewire/livewire.js`

and if you add this lines in `AppServiceProvider::register()` it should work, but it does not.

Maybe solution will be to use absolute path.

### Repository or Wirebox URL

https://github.com/lichtner/livewire-in-subdir

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by brownrl** [comment]
Did you get any where with this? I have the same problems with lievewire in a subdir, I've been trying a few things but nothing really seems to work.
